packages/core-data/src/resolvers.js

```diff
@@ -661,12 +661,42 @@ export const getUserPatternCategories =
 
 export const getNavigationFallbackId =
 	() =>
-	async ( { dispatch } ) => {
+	async ( { dispatch, select } ) => {
 		const fallback = await apiFetch( {
-			path: addQueryArgs( '/wp-block-editor/v1/navigation-fallback' ),
+			path: addQueryArgs( '/wp-block-editor/v1/navigation-fallback', {
+				_embed: true,
+			} ),
 		} );
 
+		const record = fallback?._embedded?.self;
+
 		dispatch.receiveNavigationFallbackId( fallback?.id );
+
+		if ( record ) {
+			// If the fallback is already in the store, don't invalidate navigation queries.
+			// Otherwise, invalidate the cache for the scenario where there were no Navigation
+			// posts in the state and the fallback created one.
+			const existingFallbackEntityRecord = select.getEntityRecord(
+				'postType',
+				'wp_navigation',
+				fallback?.id
```

Comment:
At this point, we know that `fallback` and its `id` are available, so I think there's no need for optional chaining.